Symfony/CS/Fixer/PSR2/TrailingSpacesFixer.php

```diff
@@ -26,7 +27,19 @@ class TrailingSpacesFixer extends AbstractFixer
     public function fix(\SplFileInfo $file, $content)
     {
         // [Structure] Don't add trailing spaces at the end of non-blank lines
-        return preg_replace('/(?<=\S)[ \t]+$/m', '', $content);
+        $contentAfterRegex = preg_replace('/(?<=\S)[ \t]+$/m', '', $content);
+
+        $originalTokens = Tokens::fromCode($content);
+        $newTokens = Tokens::fromCode($contentAfterRegex);
+
+        foreach ($newTokens as $tokenIndex => $newToken) {
+            if ($newToken->isGivenKind(array(T_ENCAPSED_AND_WHITESPACE, T_CONSTANT_ENCAPSED_STRING))) {
+                $tokenBeforeRegex = $originalTokens[$tokenIndex];
```

Comment:
this logic looks wrong to me. Can you be sure that the removal of trailing whitespaces does not change anything in the collection indexes ?